src/main/java/de/zalando/ep/zalenium/proxy/DockerSeleniumRemoteProxy.java

```diff
@@ -268,14 +315,15 @@ public void afterCommand(TestSession session, HttpServletRequest request, HttpSe
                 if (session.getSlot().getProxy() instanceof DockerSeleniumRemoteProxy) {
                     remoteName = ((DockerSeleniumRemoteProxy)session.getSlot().getProxy()).getRegistration().getContainerId();
                 }
-                ExternalSessionKey externalKey = Optional.ofNullable(session.getExternalKey()).orElse(new ExternalSessionKey("[No external key present]"));
                 LOGGER.info(String.format("Test session started with internal key %s and external key %s assigned to remote %s.",
                               session.getInternalKey(),
-                              externalKey.getKey(),
+                              session.getExternalKey().getKey(),
```

Comment:
I had it like that initially, but sometimes the browser does not start for some reason and the whole thing crashes. In those cases `session.getExternalKey()` is null and the real exception is not sent back to the client.

As a matter of fact, I just bumped into it
```
22:04:16.497 [Terminate Test Session int id: [dccfea44-3436-4b26-ab35-dd9faf2311c6] ext id: [No external key was assigned] container: [f85d9cc697a6a4ff528d5930dbfed124fe856f9998295a03126831a60293c949]] INFO  d.z.e.z.p.DockerSeleniumRemoteProxy - f85d9cc697a6a4ff528d5930dbfed124fe856f9998295a03126831a60293c949 AFTER_SESSION command received. Node should shutdown soon...
22:04:16.500 [qtp1046665075-39] INFO  o.o.g.w.s.handler.RequestHandler - Error forwarding the new session null
java.lang.NullPointerException: null
	at de.zalando.ep.zalenium.proxy.DockerSeleniumRemoteProxy.afterCommand(DockerSeleniumRemoteProxy.java:320)
	at org.openqa.grid.internal.TestSession.forward(TestSession.java:281)
	at org.openqa.grid.web.servlet.handler.RequestHandler.forwardNewSessionRequestAndUpdateRegistry(RequestHandler.java:91)
	at org.openqa.grid.web.servlet.handler.RequestHandler.process(RequestHandler.java:114)
	at org.openqa.grid.web.servlet.DriverServlet.process(DriverServlet.java:86)
	at org.openqa.grid.web.servlet.DriverServlet.doPost(DriverServlet.java:70)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
```

Could you please revert it?

Reply:
Done.